resources/js/features/games/components/TopAchieversMainRoot/TopAchieversList/TopAchieversList.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { AwardType } from '@/common/utils/generatedAppConstants';
+import { render, screen } from '@/test';
+import { createFactory } from '@/test/createFactory';
+import { createGame, createPaginatedData, createPlayerBadge, createUser } from '@/test/factories';
+
+import { TopAchieversList, TopAchieversListContainerTestId } from './TopAchieversList';
+
+export const createAchiever = createFactory<App.Platform.Data.GameTopAchiever>((faker) => {
+  return {
+    rank: 1,
+    user: createUser(),
+    score: faker.number.int({ min: 1, max: 100 }),
+    badge: null,
+  };
+});
+
+describe('Component: TopAchieversList', () => {
+  it('renders without crashing', () => {
+    // ARRANGE
+    const { container } = render<App.Platform.Data.GameTopAchieversPageProps>(
+      <TopAchieversList />,
+      {
+        pageProps: {
+          paginatedUsers: createPaginatedData([createAchiever()]),
+        },
+      },
+    );
+
+    // ASSERT
+    expect(container).toBeTruthy();
+  });
+
+  it('given there are no achievers, renders nothing', () => {
+    // ARRANGE
+    render<App.Platform.Data.GameTopAchieversPageProps>(<TopAchieversList />, {
+      pageProps: {
+        paginatedUsers: createPaginatedData([]),
+      },
+    });
+
+    // ASSERT
+    expect(screen.queryByTestId(TopAchieversListContainerTestId)).not.toBeInTheDocument();
```

Comment:
I believe this test will always pass.

I don't think the test ID is currently being used on a component. The test id has to be added to a tag, like:

```tsx
<BaseTable
  containerClassName="some stuff"
  data-testid={TopAchieversListContainerTestId}
>
```

This is all an edge case, but I'm wondering if in the event there are no players to display, rather than showing a table, we should display an `<EmptyState>` component (see: _RecentForumPosts.tsx_).

Then the test could be something like:
```tsx
// ASSERT
expect(screen.queryByRole('table')).not.toBeInTheDocument();
expect(screen.getByRole(/no top achievers/i)).toBeVisible();
```

Reply:
I changed it to look for "Rank" not being present and removed the TestID.